### PR TITLE
Fix cluster deletion with cluster name with upper case due to deleting  dns records

### PR DESCRIPTION
**Pull request by chenwany**
## Integration test:
Test test_slurm and test_exsiting_hosted_zone with cluster name uppercase, test passed, cluster deletion successfully

## Manual test:
Create cluster with name HOSTEDZONE1 with slurm
```
[ec2-user@ip-192-168-46-125 ]
$ sinfo
PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
queue1*      up   infinite      8  idle~ queue1-dy-queue1i1-[1-8]
queue1*      up   infinite      2   idle queue1-st-queue1i1-[1-2]
```

cluster hosted name: hostedzone1.pcluster
Records:
there are 2 records in the 
```
Record name | Type | Routing policy | Differentiator | Value/Route traffic to
-- | -- | -- | -- | --
  | hostedzone1.pcluster | NS | Simple | - | ns-1536.awsdns-00.co.uk.ns-0.awsdns-00.com.ns-1024.awsdns-00.org.ns-512.awsdns-00.net.
  | hostedzone1.pcluster | SOA | Simple | - | ns-1536.awsdns-00.co.uk. awsdns-hostmaster.amazon.com. 1 7200 900 1209600 86400
  | queue1-st-queue1i1-1.hostedzone1.pcluster | A | Simple | - | 192.168.45.77
  | queue1-st-queue1i1-2.hostedzone1.pcluster | A | Simple | - | 192.168.62.183
```

- ssh into the compute: 
```
[ec2-user@ip-192-168-46-125 ~]$ ssh queue1-st-queue1i1-1
```
-cat /etc/resolv.conf
```
# Generated by NetworkManager
search us-west-1.compute.internal hostedzone1.pcluster
nameserver 192.168.0.2
options timeout:2 attempts:5
```
- submit job run succesfully

- Delete cluster:
```
cluster can be deleted succesfully:
START RequestId: 37b33c37-e755-4c4d-99cb-272fc865f2ed Version: $LATEST
--
[INFO]	2021-09-02T16:46:31.016Z	37b33c37-e755-4c4d-99cb-272fc865f2ed	Deleting DNS records from Z09847202JG2GWZ5YK0PH
[INFO]	2021-09-02T16:46:32.052Z	37b33c37-e755-4c4d-99cb-272fc865f2ed	Deleting ResourceRecordSets end with hostedzone1.pcluster.
[INFO]	2021-09-02T16:46:32.506Z	37b33c37-e755-4c4d-99cb-272fc865f2ed	Sleeping for 5 seconds before retrying DNS records deletion.
[INFO]	2021-09-02T16:46:37.521Z	37b33c37-e755-4c4d-99cb-272fc865f2ed	DNS records deletion from Z09847202JG2GWZ5YK0PH: COMPLETED
[INFO]	2021-09-02T16:46:37.522Z	37b33c37-e755-4c4d-99cb-272fc865f2ed	PhysicalResourceId present in event, Using that for response
[INFO]	2021-09-02T16:46:37.691Z	37b33c37-e755-4c4d-99cb-272fc865f2ed	CloudFormation returned status code: OK
```

### CODE REVIEW

File: tests/integration-tests/tests/dns/test_dns.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def test_existing_hosted_zone(
 
     # Test domain name matches expected domain name
     resolv_conf = remote_command_executor.run_remote_command("cat /etc/resolv.conf").stdout
-    assert_that(resolv_conf).contains(cluster.cfn_name + "." + domain_name)
```
Comment (enrico-usai) [inline tests/integration-tests/tests/dns/test_dns.py:101]: let's extend an integration tests to create a cluster with an UPPERCASE cluster name, in this way we're sure all these changes are working as expected.

Comment (demartinofra) [reply tests/integration-tests/tests/dns/test_dns.py:101]: we could also modify the current test to create a cluster with an uppercase letter. please leave the integ-tests prefix and the suffix unaltered

Comment (chenwany) [reply tests/integration-tests/tests/dns/test_dns.py:101]: Good idea! Modify cluster_factoryAdd 2 integration test test_slurm and test_existing_hosted_zone to include uppercase in clustername

Comment (demartinofra) [reply tests/integration-tests/tests/dns/test_dns.py:101]: awesome thanks

**Review by demartinofra — APPROVED** [review]

**Review by enrico-usai — APPROVED** [review]
When inserting a new parameter in the fixture we should consider extensibility.
This parameter `upper_case_cluster_name` can only be used for having a cluster with uppercase letters.

Introducing for example a parameter called `cluster_name_infix` would permit us to test different use cases in future, e.g. with other characters in the middle or other combination that might break our code.

Anyway let's go ahead by merging the patch. Thanks!

**Pull request merged**